src/io.jl

```diff
@@ -140,6 +140,84 @@ function convert_to_internal(
 end
 
 #------------------------------------------------------------------------------
+function convert_to_internal(
+        orig_polys::Vector{T},
+        ordering::Symbol) where {T<:SIMDPolynomials.AbstractPolynomial}
+
+    isempty(orig_polys) && error("Empty input")
+    ordering in (:input, :lex, :degrevlex, :deglex) || error("Not supported ordering $ordering")
+
+    ring = extract_ring(orig_polys)
+    exps, cfs = extract_polys(ring, orig_polys)
+    assure_ordering!(ring, ordering, exps, cfs)
+    ring, exps, cfs
+end
+
+function extract_ring(orig_polys::Vector{<:SIMDPolynomials.AbstractPolynomial})
+    f = first(orig_polys)
+
+    nv = maximum(SIMDPolynomials.nvariables, orig_polys)
```

Comment:
I am not sure if taking maximum is correct. What if the system is [x, y] ?

Reply:
Yeah, it's correct only for PackedMonomial, and not true in general.